src/main/java/seedu/address/storage/JsonAdaptedPet.java

```diff
@@ -47,35 +57,65 @@ public JsonAdaptedPet(ReadOnlyPet source) {
      * @throws IllegalValueException if there were any data constraints violated in the adapted
      *     task.
      */
-    public ReadOnlyPet toModelType() throws IllegalValueException {
-        // TODO set up proper model for all attributes of the Pet
-        // if (name == null) {
-        // throw new IllegalValueException(
-        // String.format(MISSING_FIELD_MESSAGE_FORMAT, "name"));
-        // }
-        // if (!Name.isValidName(name)) {
-        // throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
-        // }
-        // final Name modelName = new Name(name);
-
-        // if (exp == null) {
-        // throw new IllegalValueException(
-        // String.format(MISSING_FIELD_MESSAGE_FORMAT, "exp"));
-        // }
-        // if (!exp.isValidexp(exp)) {
-        // throw new IllegalValueException(exp.MESSAGE_CONSTRAINTS);
-        // }
-        // final exp modelexp = new exp(priority);
-
-        // if (level == null) {
-        // throw new IllegalValueException(
-        // String.format(MISSING_FIELD_MESSAGE_FORMAT, level.class.getSimpleName()));
-        // }
-        // if (!level.isValidlevel(level)) {
-        // throw new IllegalValueException(level.MESSAGE_CONSTRAINTS);
-        // }
-        // final level modellevel = new level(level);
-
-        return new Pet(name, exp, level, mood, lastDoneTaskTime);
+    public ReadOnlyPet toModelType() throws IllegalValueException, InvalidPetException {
+        if (name == null) {
```

Comment:
This method looks a bit big, would it be a good idea to split up the control flow to it's own methods and return the final pet?